OpenRA.Test/OpenRA.Game/VariableExpressionTest.cs

```diff
@@ -333,7 +344,120 @@ public void TestParseErrors()
 			AssertParseFailure("false ||", "Missing value or sub-expression at end for `||` operator");
 			AssertParseFailure("1 <", "Missing value or sub-expression at end for `<` operator");
 			AssertParseFailure("-1a", "Number -1 and variable merged at index 0");
+		}
+
+		[TestCase(TestName = "Test hyphen parser errors")]
+		public void TestParseHyphenErrors()
+		{
 			AssertParseFailure("-", "Missing value or sub-expression at end for `-` operator");
+			AssertParseFailure("-1-1", "Missing binary operation before `-1` at index 2");
+			AssertParseFailure("5-1", "Missing binary operation before `-1` at index 1");
+			AssertParseFailure("6 -1", "Missing binary operation before `-1` at index 2");
+			AssertParseFailure("t -1", "Missing binary operation before `-1` at index 2");
+		}
+
+		[TestCase(TestName = "Test mixed charaters at end of identifier parser errors")]
+		public void TestParseMixedEndErrors()
+		{
+			AssertParseFailure("t- 1", "Invalid identifier end character at index 1 for `t-`");
+			AssertParseFailure("t-", "Invalid identifier end character at index 1 for `t-`");
+			AssertParseFailure("t. 1", "Invalid identifier end character at index 1 for `t.`");
+			AssertParseFailure("t.", "Invalid identifier end character at index 1 for `t.`");
+			AssertParseFailure("t@ 1", "Invalid identifier end character at index 1 for `t@`");
+			AssertParseFailure("t@", "Invalid identifier end character at index 1 for `t@`");
+			AssertParseFailure("t$ 1", "Invalid identifier end character at index 1 for `t$`");
+			AssertParseFailure("t$", "Invalid identifier end character at index 1 for `t$`");
+		}
+
+		[TestCase(TestName = "Test binary operator whitespace parser errors")]
+		public void TestParseSpacedBinaryOperatorErrors()
+		{
+			// Hyphen creates other parsing errors for `6 -1`, `6-1`, `t -1`, and `t- 1` and `t-1` is valid.
```

Comment:
My brain is failing to parse this sentence.  Can you split this into two sentences so that it is clear which cases are parsing errors and which are valid?